File: TeacherStudentService/src/main/java/com/student/dtos/StudentDto.java
package com.student.dtos;

import java.time.LocalDate;
import com.fasterxml.jackson.annotation.JsonProperty; // Required for mapping
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PastOrPresent;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class StudentDto {

    @NotBlank(message = "Student name is required")
    @JsonProperty("Name") // Maps .NET 'Name' to 'studentName'
    private String studentName;

    @NotBlank(message = "Email is required")
    @Email(message = "Invalid email format")
    @JsonProperty("Email") // Maps .NET 'Email' to 'email'
    private String email;

    @NotBlank(message = "Password is required")
    @JsonProperty("Password")
    private String password;

    @NotNull(message = "Admission date is required")
    @PastOrPresent(message = "Admission date cannot be in future")
    @JsonProperty("AdmissionDate") // Maps .NET 'AdmissionDate' to 'admissionDate'
    private LocalDate admissionDate;
    
    @JsonProperty("Role")
    private String role;

}

File: UserAuthService/DTOs/StudentDTO.cs
using System.Text.Json.Serialization; // Required for mapping names

namespace UserAuthService.DTOs
{
    public class StudentDTO
    {
        // Java: private String studentName;
        [JsonPropertyName("studentName")]
        public string StudentName { get; set; } = string.Empty;

        // Java: private String email;
        [JsonPropertyName("email")]
        public string Email { get; set; } = string.Empty;

        // Java: private String password;
        [JsonPropertyName("password")]
        public string Password { get; set; } = string.Empty;

        

        
    }
}